config/routes.rb

```diff
@@ -3,6 +3,7 @@
 Rails.application.routes.draw do
   resources :challenges
   get 'challenges/:id/entries' => 'challenges#entries'
+  post 'submissions/:id/soft_delete' => 'submissions#soft_delete'
```

Comment:
Add `, as: :submission_soft_delete` to add a URL helper. This should also be placed under (but not in) the `resources :submission` block to keep similar routes grouped together (I clean up that stuff a lot in the API PR).